driver-core/src/main/java/com/datastax/driver/core/EventDebouncer.java

```diff
@@ -151,6 +155,7 @@ private void completeAllPendingFutures() {
 
         if (state == State.RUNNING) {
             int count = eventCount.get();
+            int maxPendingEvents = maxPendingEvents();
```

Comment:
To follow up on my previous comment about maxPendingEvents, the issue is not only when it changes while the method is executing, but also when it changes between calls, for example:

* maxPendingEvents = 100, 49 events are enqueued
* eventReceived is called, count increments to 50 => we schedule a delayed delivery
* maxPendingEvents is changed to 50
* eventReceived is called, count increments to 51 => we do nothing

In this scenario we go over the max without ever triggering the condition `(count == maxPendingEvents)`. As I noted before a scheduled delivery is still planned, so maybe we can overlook this as an edge case.